flask_oidc/__init__.py

```diff
@@ -187,11 +196,115 @@ def init_app(self, app):
         except KeyError:
             pass
 
-    def load_secrets(self, app):
-        # Load client_secrets.json to pre-initialize some configuration
-        return _json_loads(open(app.config['OIDC_CLIENT_SECRETS'],
-                                   'r').read())
-        
+    def init_provider(self, provider):
+        """
+        Do setup for a specific provider
+
+        :param provider: The provider to initialize.
+        :type provider: Dictionary with at lease 'base_url' item
+        """
+
+        secrets = self.load_secrets(provider)
+        assert secrets != None, "Problem with loading secrets"
+
+        self.client_secrets = list(secrets.values())[0]
+        secrets_cache = DummySecretsCache(secrets)
+
+        # Initialize oauth2client
+        self.flow = flow_from_clientsecrets(
+            current_app.config['OIDC_CLIENT_SECRETS'],
+            scope=current_app.config['OIDC_SCOPES'],
+            cache=secrets_cache)
+
+        assert isinstance(self.flow, OAuth2WebServerFlow)
+
+        current_app.config['OIDC_VALID_ISSUERS'] = (self.client_secrets.get('issuer') or GOOGLE_ISSUERS)
+
+    def load_secrets(self, provider):
+
+        try:
+            static_secrets = current_app.config.get('OIDC_CLIENT_SECRETS', None)
```

Comment:
Maybe return an error if `provider` is not `None` in this case, to avoid confusing people?